hackathon_site/dashboard/frontend/src/slices/event/teamDetailSlice.ts

```diff
@@ -80,6 +120,26 @@ const teamDetailSlice = createSlice({
             state.isLoading = false;
             state.error = payload?.message ?? "Something went wrong";
         });
+        builder.addCase(updateParticipantIdProvided.pending, (state) => {
+            state.isLoading = true;
```

Comment:
you are right about the loading states, you should create a new loading state just for clicking the checkbox and update that here instead.

Reply:
added new loading state